posthog/tasks/alerts/test/test_checks.py

```diff
@@ -0,0 +1,167 @@
+import pytest
+from typing import Any, Optional
+from unittest.mock import MagicMock, patch
+
+from freezegun import freeze_time
+
+from posthog.models.instance_setting import set_instance_setting
+from posthog.tasks.alerts.checks import check_all_alerts
+from posthog.test.base import APIBaseTest, _create_event, flush_persons_and_events, ClickhouseDestroyTablesMixin
+from posthog.api.test.dashboards import DashboardAPI
+from posthog.schema import ChartDisplayType, EventsNode, TrendsQuery, TrendsFilter
+from posthog.tasks.test.utils_email_tests import mock_email_messages
+
+
+@freeze_time("2024-06-02T08:55:00.000Z")
+@patch("posthog.tasks.alerts.checks.EmailMessage")
+class TestDetectAlertsAnomaliesTasks(APIBaseTest, ClickhouseDestroyTablesMixin):
+    def setUp(self) -> None:
+        super().setUp()
+        set_instance_setting("EMAIL_HOST", "fake_host")
+        set_instance_setting("EMAIL_ENABLED", True)
+        self.settings(CELERY_TASK_ALWAYS_EAGER=True)
+        self.dashboard_api = DashboardAPI(self.client, self.team, self.assertEqual)
```

Comment:
I'm not a fan of using API to test task behavior in task tests... as this couple theses tests to API implementation. Can we just use the model? (Or do we do this like that everywhere in PostHog 🙈?)

Reply:
there's more than one model involved in a dashboard, so this couples the tests to a single API regardless of what's beneath it... in this case the API since otherwise you can (and we did) break the API and not know about it because you're not using it

Reply:
I prefer to test using public interfaces as it's usually much closer to what happens in production. Less importantly, it covers more functionality and usually simpler to write.

> as this couple theses tests to API implementation

I'm not sure it's a bad thing, see my point about much closer to production.